openpower_guard_interface.cpp

```diff
@@ -20,141 +21,71 @@ namespace FileError = sdbusplus::xyz::openbmc_project::Common::File::Error;
 namespace HardwareIsolationError =
     sdbusplus::xyz::openbmc_project::HardwareIsolation::Error;
 
+/**
+ * @brief Helper MACRO to call libguard interface and to have
+ *        common exception handler
+ *
+ * @param[in] INTERFACE_SIGNATURE - To replace inside try block
+ */
+#define CALL_LIBGUARD_INTERFACE(INTERFACE_SIGNATURE)                           \
+    try                                                                        \
+    {                                                                          \
+        INTERFACE_SIGNATURE                                                    \
+    }                                                                          \
+    catch (libguard::exception::GuardFileOpenFailed & e)                       \
+    {                                                                          \
+        throw FileError::Open();                                               \
+    }                                                                          \
+    catch (libguard::exception::GuardFileReadFailed & e)                       \
+    {                                                                          \
+        throw FileError::Read();                                               \
+    }                                                                          \
+    catch (libguard::exception::GuardFileWriteFailed & e)                      \
```

Comment:
Catch **GuardFileSeekFailed** as well which is added recently in libguard

Reply:
InvalidEntry exception is added for internal purpose not impact from application side so, just adding `GuardFileSeekFailed`.